homeassistant/components/switch/__init__.py

```diff
@@ -68,9 +74,46 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     )
     await component.async_setup(config)
 
-    component.async_register_entity_service(SERVICE_TURN_OFF, {}, "async_turn_off")
-    component.async_register_entity_service(SERVICE_TURN_ON, {}, "async_turn_on")
-    component.async_register_entity_service(SERVICE_TOGGLE, {}, "async_toggle")
+    async def async_handle_switch_off_service(
+        switch: SwitchEntity, call: ServiceCall
+    ) -> None:
+        """Handle turning off a light."""
+        # pylint: disable-next=protected-access
+        if switch.is_on:
+            switch.async_set_context_filter(state_filter(STATE_OFF))
+        else:
+            switch.async_set_context_filter(async_false_filter)
```

Comment:
Why set it to something at all? Shouldn't the context filter just expire when the context does?

Reply:
It's to prevent any state changes to be attributed to the service call